floss/main.py

```diff
@@ -592,65 +592,58 @@ def create_ida_script_content(sample_file_path, decoded_strings, stack_strings):
             main_commands.append("AppendLvarComment(%d, %d, \"FLOSS stackstring: %s\", True)" % (ss.fva, ss.frame_offset, sanitized_string))
             ss_len += 1
-    main_commands.append("print \"Imported stackstrings from FLOSS\"")
+    main_commands.append("print(\"Imported stackstrings from FLOSS\")")
 
-    script_content = """from idc import RptCmt, Comment, MakeRptCmt, MakeComm, GetFrame, GetFrameLvarSize, GetMemberComment, SetMemberComment, Refresh
+    script_content = """get_cmt, set_cmt, get_func_attr, get_member_cmt, set_member_cmt
 
 
-def AppendComment(ea, s, repeatable=False):
-    # see williutils and http://blogs.norman.com/2011/security-research/improving-ida-analysis-of-x64-exception-handling
-    if repeatable:
-        string = RptCmt(ea)
-    else:
-        string = Comment(ea)
+def AppendComment(ea, string, repeatable=False):
+    current_string = get_cmt(ea, repeatable)
 
-    if not string:
-        string = s  # no existing comment
+    if not current_string:
+        cmt = string
     else:
-        if s in string:  # ignore duplicates
+        if string in current_string:  # ignore duplicates
             return
-        string = string + "\\n" + s
-    if repeatable:
-        MakeRptCmt(ea, string)
-    else:
-        MakeComm(ea, string)
+        cmt = string + "\\n" + string
+    set_cmt(ea, cmt, repeatable)
 
 
 def AppendLvarComment(fva, frame_offset, s, repeatable=False):
-    stack = GetFrame(fva)
+    stack = get_func_attr(fva, FUNCATTR_FRAME)
     if stack:
-        lvar_offset = GetFrameLvarSize(fva) - frame_offset
+        lvar_offset = get_func_attr(fva, FUNCATTR_FRSIZE) - frame_offset
         if lvar_offset and lvar_offset > 0:
-            string = GetMemberComment(stack, lvar_offset, repeatable)
+            string = get_member_cmt(stack, lvar_offset, repeatable)
             if not string:
                 string = s
             else:
                 if s in string:  # ignore duplicates
                     return
                 string = string + "\\n" + s
-            if SetMemberComment(stack, lvar_offset, string, repeatable):
-                print "FLOSS appended stackstring comment \\\"%%s\\\" at stack frame offset 0x%%X in function 0x%%X" %% (s, frame_offset, fva)
+            if set_member_cmt(stack, lvar_offset, string, repeatable):
+                print("FLOSS appended stackstring comment \\\"%%s\\\" at stack frame offset 0x%%X in function 0x%%X" %% (s, frame_offset, fva))
                 return
-    print "Failed to append stackstring comment \\\"%%s\\\" at stack frame offset 0x%%X in function 0x%%X" %% (s, frame_offset, fva)
+    print("Failed to append stackstring comment \\\"%%s\\\" at stack frame offset 0x%%X in function 0x%%X" %% (s, frame_offset, fva))
 
 
 def main():
-    print "Annotating %d strings from FLOSS for %s"
+    print("Annotating %d strings from FLOSS for %s")
     %s
-    Refresh()
+    ida_kernwin.refresh_idaview_anyway
```

Comment:
should this be a function call?

Reply:
yes 👍 👍 it was failing, but It didn't raise an error in IDA and I hadn't realized